### PR TITLE
ocm: added new policy set builder and unit tests

### PR DESCRIPTION
**Pull request by klaskosk**
Added `NewPolicySetBuilder` as well as a method to add a new `Policy` to the set. Tests for the new methods are included.

### CODE REVIEW

**Review by kononovn — CHANGES_REQUESTED** [review]
Re-base is needed.

**Comment by klaskosk** [comment]
@kononovn Rebased

File: pkg/ocm/policyset_test.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+			assert.Nil(t, err)
+			assert.Equal(t, testBuilder.Definition.Name, policySetBuilder.Definition.Name)
+			assert.Equal(t, testBuilder.Definition.Spec.Description, policySetBuilder.Definition.Spec.Description)
+		} else {
+			assert.NotNil(t, err)
+		}
+	}
+}
+
+func TestWithPolicy(t *testing.T) {
```
Comment (kononovn) [inline pkg/ocm/policyset_test.go:302]: Please re-name to  TestPolicySetWithPolicy.

**Review by kononovn — CHANGES_REQUESTED** [review]
looks good, Just few comment to fix.

**Comment by klaskosk** [comment]
@kononovn Thanks, I've made those changes

**Review by kononovn — APPROVED** [review]
/lgtm

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**